File: src/features/typing/data/go/partition_labels.go
package main

func partitionLabels(s string) []int {
	last := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		last[s[i]] = i
	}
	result := []int{}
	start, end := 0, 0
	for i := 0; i < len(s); i++ {
		if last[s[i]] > end {
			end = last[s[i]]
		}
		if i == end {
			result = append(result, i-start+1)
			start = i + 1
		}
	}
	return result
}

File: src/features/typing/data/dart/partition_labels.dart
class Solution {
  List<int> partitionLabels(String s) {
    final last = List<int>.filled(128, 0);
    for (int i = 0; i < s.length; i++) {
      last[s.codeUnitAt(i)] = i;
    }
    final result = <int>[];
    var start = 0;
    var end = 0;
    for (int i = 0; i < s.length; i++) {
      if (last[s.codeUnitAt(i)] > end) {
        end = last[s.codeUnitAt(i)];
      }
      if (i == end) {
        result.add(i - start + 1);
        start = i + 1;
      }
    }
    return result;
  }
}
